fv3config/_config.py

```diff
@@ -60,12 +78,20 @@ def config_from_namelist(namelist_filename):
         InvalidFileError: if the specified filename does not exist
     """
     try:
-        return_dict = f90nml.read(namelist_filename)
+        return_dict = _to_nested_dict(f90nml.read(namelist_filename).items())
```

Comment:
Would [f90nml.Namelist.todict](https://f90nml.readthedocs.io/en/latest/usage.html#f90nml.namelist.Namelist.todict) work here? If so, then you could delete `_to_nested_dict`.

Reply:
I think this could be done by registering a `yaml` serializer for orderdicts, but that is probably more complicated than this approach TBH.

Reply:
Ok. I guess it's okay. Although `yaml` does have a way to register a specific processor for data types, I think this approach is cleaner.